datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -342,6 +342,53 @@ impl LogicalPlan {
         self.accept(&mut visitor)?;
         Ok(visitor.using_columns)
     }
+
+    pub fn clone_with_inputs(&self, inputs: Vec<LogicalPlan>) -> Result<LogicalPlan, DataFusionError> {
```

Comment:
This has a non trivial amount in common with `from_plan`: 
https://github.com/apache/arrow-datafusion/blob/4cb8ac094ee88dc0023b4cde1b39840a0a362ee0/datafusion/expr/src/utils.rs#L363


I wonder if we could make `from_plan` easier to find / use / more general 🤔